### DISCUSSION TITLE
zinit self-update fails with error `fatal: couldn't find remote ref master`, after migration

### DISCUSSION THREAD

**Discussion by montaropdf** [zdharma-continuum/zinit#366] [Q&A] [answered]
Hello,

I just migrated my 2 (or more) years old zinit installation, everything seems to work proprelly, except for the `self-update` and `status` commands, returning the error `fatal: couldn't find remote ref master`.

While in `~/.zinit/bin`, `git branch` displays a `main` and `master` branches, the latter being the current branch.

The command `update` does not trigger any error and update the plugins, though.

**Comment by pschmitt** [comment, marked as answer]
The easiest way out if this is probably to delete the zinit git dir and clone again..

```zsh
rm -rf $ZINIT[BIN_DIR]
git clone https://github.com/zdharma-continuum/zinit $ZINIT[BIN_DIR]
```

**Reply by montaropdf** [reply]
After cloning, did I just need to log off and on again?

Or, do I need, first, to perform a zinit self-update?

**Reply by pschmitt** [reply]
I'd just `exec zsh` in the current shell.

**Reply by vladdoster** [reply]
@montaropdf, @pschmitt's suggestion will work, but here is an onliner for brevity...

```zsh
rm -rf "$ZINIT[BIN_DIR]" \
&& git clone https://github.com/zdharma-continuum/zinit "$ZINIT[BIN_DIR]" \
&& exec zsh -il
```

**Reply by montaropdf** [reply]
> @montaropdf, @pschmitt's suggestion will work, but here is an onliner for brevity...
> 
> ```shell
> rm -rf "$ZINIT[BIN_DIR]" \
> && git clone https://github.com/zdharma-continuum/zinit "$ZINIT[BIN_DIR]" \
> && exec zsh -il
> ```

Your one-liner works. I perform a `zinit self-update` that reports everything is up to date.

So this case can be closed.

Thanks you, all, for your help.
